### DISCUSSION TITLE
A complete executatble Zim Desktop Wiki that could run without installing locally ...

### DISCUSSION THREAD

**Discussion by rajibando** [zim-desktop-wiki/zim-desktop-wiki#2601] [Ideas]
My earlier discussion on linking a phrase in one page of a Zim Desktop Wiki top level file to another phrase of another page of the same file led me to this insight.

There are many such applications that are presently available, Like GeoGebra, that could simply be downloaded and run without installation and there isn't very system specific so long as the OS remains the same and isn't too old to not to be used.

Even an online version of GeoGebra is available from the Browser but is a bit taxing for the local system to execute.

The latest version is in Debian `trixie` and `sid`.

I have downloaded the zim edition in `trixie` and a priliminary test says all dependencies are satisfied. Let's see.

So with this idea a complete updated version could be immediately be downloaded and used without upgrading the `depends`, `recommends`, `suggests`, `enhances` packages as far as possible.

**Comment by dehesselle** [comment]
Zim can already do that: just download the source and run. It is documented in the [Installing from Source](https://github.com/zim-desktop-wiki/zim-desktop-wiki?tab=readme-ov-file#installing-from-source) section in the readme.

**Reply by rajibando** [reply]
Thank you, Mr. de Hesselle, for replying to my post.

Actually I was confused by the usage of the term 'Source'. When a fully complete program is available for downloads why should users download and then compile/build from source?

As I had said earlier,  we need just to download the program and execute it. Like in GeoGebra.

So the phrase, **Installing from Source**, left me confused. Which is why I downloaded the package from Debian Repository of Trixie and then checked and installed it like it is done in Debian. The heading should have said, **Running from Source**.

But thank you once more for informing me. Please consider changing the word **Installing** on the concerned page to **Running**.

**Reply by dehesselle** [reply]
Well, it's Python - the source _is_ the program. But if you're not familiar with that, I can understand it might not be the first thing that comes to mind.

There is an overlap between running and installing from source, that's why it's a combined section. The very first paragraph begins with

> **NOTE:** You don't need to install Zim [...]

Did you overlook that on your first read or do you find it confusing nevertheless?

**Comment by rajibando** [comment]
Thank you, Mr. de Hesselle, for again replying to my post so promptly.
Yes, truly I don't know much about Python rather than some simple python codes.
I have been using Knoppix from 2006 and Debian from 2008. In the Debian environment we usually did _make_ and _make install_ with makefiles from the source packages, which I rarely did.
Programming or developing was never my profession. I learnt some aspects of computing simply out of curiosity and need and I self-learnt what I had to, along the way.
Apologise to have offended you. I didn't overlook the first line with _NOTE:_. I assumed that since English isn't necessarily the first language of many programmers, it was an error in sentence composition for the apparently conflicting two phrases, _Installing from Source_ and _You don't need to install zim in order to test it. You should be able to run it directly from the source directory ..._
By my experience of two decades I have come to note that Documentation is one issue that Open Source Software is the weakest in. Open Source programmers do outrageously wonderful programming, some times magical, but fail to document those unique functionalities.
In Zim too, the usage of Anchor isn't well-documented.
Brilliant programmers fail to register that the users might not be as gifted as they are in using a functionality like they do.
My bad.
So I stand corrected.
Thank you, Mr. de Hesselle for instructing me on the issue.
